tests/test_plotfileparticledata.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+
+import random
+import shutil
+from dataclasses import dataclass
+
+import numpy as np
+import pytest
+
+import amrex.space3d as amr
+
+# Particle container constructor -- depending on if gpus are available use the
+# GPU-enabled version. This uses the FHDeX Particle Container for the test
+if amr.Config.have_gpu:
+    PCC = amr.ParticleContainer_16_4_0_0_managed
+else:
+    PCC = amr.ParticleContainer_16_4_0_0_default
+
+
+@dataclass
+class Particle:
+    """
+    Helper class to work with particle data
+    """
+
+    x: float
+    y: float
+    z: float
+
+    idx: int
+
+    def to_soa(self, aos_numpy):
+        """Fill amr particle SoA with x, y, z, idx data"""
+        aos_numpy["x"] = self.x
+        aos_numpy["y"] = self.y
+        aos_numpy["z"] = self.z
+        aos_numpy["idata_0"] = self.idx
+
+
+def generate_test_particles(n_part):
+    """
+    Returns a list of test particles scattered throught the domain
+    """
+    particles = list()
+
+    def generator():
+        return 1 - 2 * random.random()
+
+    for i in range(n_part):
+        particles.append(Particle(x=generator(), y=generator(), z=generator(), idx=i))
+
+    return particles
+
+
+def particle_container(Rpart, std_geometry, distmap, boxarr, std_real_box):
+    """
+    Generate a fresh particle container, containing copies from Rpart
+    """
+    pc = PCC(std_geometry, distmap, boxarr)
+
+    iseed = 1
+    myt = amr.ParticleInitType_16_4_0_0()
+    pc.init_random(len(Rpart), iseed, myt, False, std_real_box)
+
+    particles_tile_ct = 0
+    # assign some values to runtime components
+    for lvl in range(pc.finest_level + 1):
+        for pti in pc.iterator(pc, level=lvl):
+            aos = pti.aos()
+            aos_numpy = aos.to_numpy(copy=False)
+            for i, p in enumerate(aos_numpy):
+                Rpart[i + particles_tile_ct].to_soa(p)
+            particles_tile_ct += len(aos_numpy)
+
+    pc.redistribute()
```

Comment:
@JBlaschke please double check (maybe with `print` debugging) if this does what you expect to do or if you geometry needs adjustment, if the `pc.redistribute` removes particles.